django/template/base.py

```diff
@@ -841,12 +840,12 @@ def __repr__(self):
     def render(self, context):
         try:
             output = self.filter_expression.resolve(context)
+            return render_value_in_context(output, context)
         except UnicodeDecodeError:
             # Unicode conversion can fail sometimes for reasons out of our
             # control (e.g. exception rendering). In that case, we fail
             # quietly.
             return ''
-        return render_value_in_context(output, context)
```

Comment:
I forgot to mention, I fixed this to no longer move ``render_value_in_context``. I tracked the ``UnicodeDecodeError`` to it's source in ``Template.get_exception_info`` and caught it there instead. This way, we're sure never to pass an invalid unicode string to the debug page.